src/de.rs

```diff
@@ -70,15 +90,15 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Depythonizer<'de> {
             self.deserialize_bool(visitor)
         } else if obj.is_instance::<PyByteArray>()? || obj.is_instance::<PyBytes>()? {
             self.deserialize_bytes(visitor)
-        } else if obj.is_instance::<PyDict>()? {
+        } else if P::Dict::is_instance(obj)? {
             self.deserialize_map(visitor)
         } else if obj.is_instance::<PyFloat>()? {
             self.deserialize_f64(visitor)
         } else if obj.is_instance::<PyFrozenSet>()? {
             self.deserialize_tuple(obj.len()?, visitor)
         } else if obj.is_instance::<PyInt>()? {
             self.deserialize_i64(visitor)
-        } else if obj.is_instance::<PyList>()? {
+        } else if P::List::is_instance(obj)? {
             self.deserialize_tuple(obj.len()?, visitor)
```

Comment:
... continuing the above, I think we get something like this here:

```suggestion
        } else if let Ok(seq) = obj.downcast::<&PySequence>()? {
            if let Ok(len) = seq.len() {
                self.deserialize_tuple(obj.len()?, visitor)
            else {
                self.deserialize_seq(visitor)
            }
        }
```